src/azure-cli-testsdk/azure/cli/testsdk/utilities.py

```diff
@@ -210,6 +211,38 @@ def process_response(self, response):
         return response
 
 
+class EmailAddressReplacer(RecordingProcessor):
+    """Replace email address like xxx@microsoft.com with test@example.com"""
+
+    EMAIL_REPLACEMENT = 'test@example.com'
+
+    def _replace_email_address(self, text):
+        pattern = r'[\w.%#+-]+[%40|@|_]microsoft.com'
```

Comment:
For `test_microsoft.com#EXT#@AzureSDKTeam.onmicrosoft.com`, this regex only replaces the `test_microsoft.com` part. Should we consider replacing the full string `test_microsoft.com#EXT#@AzureSDKTeam.onmicrosoft.com` to avoid showing the tenant domain too?

Reply:
If I loose the regex to recognize all string in the format of `xxx@xxx.xx`, then `test_microsoft.com#EXT#@AzureSDKTeam.onmicrosoft` can be replaced, but there will be many mismatch for example `ssh@10.172` and the recording file can't work.

I'm thinking that as long as the microsoft part is replaced, then the full string will be invalid so I restricted the pattern to recognize `[@|%40|_]microsoft.com` only